### DISCUSSION TITLE
Redirect server log to Docker logs

### DISCUSSION THREAD

**Discussion by clabnet** [Kitware/trame#632] [Q&A]
Hi, 
it is difficult to debug a dockerized trame server app, because cannot be simple to enter into the docker container to see the `deploy\server\logs\launcher` folder, many time is impossible (Azure for example). 
Is there a mode to redirect the launcher logs to Docker container logs ?
Thank you in advance.

**Comment by jourdain** [comment]
Why don't you mount the log directory on a drive you can look at?

The problem with the redirect is when you have multiple session running the output would be intermixed.

**Reply by krishanbhasin-px** [reply]
Is it possible to expose the session ID to the Trame code? Then the logs could include that - when they're all redirected to STDOUT and shipped to a logging aggregator, they could be filtered there to just the specific session, bypassing the intermingling problem?

**Reply by jourdain** [reply]
Yes that could be done, but if all of that is for a Kubernete deployment, you are solving the wrong problem.

**Reply by krishanbhasin-px** [reply]
Could you clarify what you mean by solving the wrong problem? I'd like to deploy my trame app into a kubernetes cluster so I can have colleagues use it

**Reply by jourdain** [reply]
What I mean is that kubernetes allow you do a better job at scaling your jobs across a cluster. Ideally, you want each of the trame job to be scheduled by kubernete and allocated to a given hardware based on the load and need. With such setup, you don't have the logging issue mentioned above as your are skipping the multi-session part and ideally you get a better hardware utilization across machines. The multi-user trame docker image on the other hand is really designed for a single machine with its hardware limitation. Basically, I would rather put some effort in building a native kubernete deployment than patching something that is not designed for it. 

Side note: you can create a process that watch those logs and just print each lines with the appropriate session id (file name) in a fairly easy manner.
